src/ray/core_worker/transport/thread_pool.h

```diff
@@ -53,12 +54,6 @@ class BoundedExecutor {
   void Join();
 
  private:
-  bool ThreadsAvailable() EXCLUSIVE_LOCKS_REQUIRED(mu_);
-
-  /// Protects access to the counters below.
-  absl::Mutex mu_;
-  /// The number of currently running tasks.
-  int num_running_ GUARDED_BY(mu_);
   /// The max number of concurrently running tasks allowed.
   const int max_concurrency_;
```

Comment:
Can we delete `max_concurrency_` as well? Or should we remove `BoundedExecutor` and use `boost::asio::thread_pool` directly?

Reply:
Deleting max_concurrency_ is ok to me.

Removing `BoundedExecutor` is hard to remove due to we use a ConcurrencyGroupManager to unify async actor and threaded actor. `BoundedExecutor` is a specific type of ExecutorType, which is the generic type of `ConcurrencyGroupManager`. See https://github.com/ray-project/ray/blob/master/src/ray/core_worker/transport/concurrency_group_manager.h

Reply:
I see